server/mem_agent/engine.py

```
@@ -332,3 +350,656 @@ def _subprocess_entry() -> None:
+import sys
+import os
+
+# Restore session state safely via pickle
+_state_b64 = "{state_b64}"
+_session_state = pickle.loads(base64.b64decode(_state_b64))
+
+# Make session state available as globals
+globals().update(_session_state)
+
+# Change to workspace
+os.chdir({repr(self.workspace_path)})
```

Ruff (0.14.13)

Use explicit conversion flag

Replace with conversion flag

(RUF010)